src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets

```diff
@@ -343,10 +343,23 @@ Copyright (c) .NET Foundation. All rights reserved.
     
   </Target>
 
-  <Target Name="CheckWindowsDesktopSdkInUse" 
-          BeforeTargets="_CheckForInvalidConfigurationAndPlatform" 
-          Condition="'$(UseWpf)' == 'true' Or '$(UseWindowsForms)' == 'true'">
-    <NetSdkWarning Condition="'$(_MicrosoftNetSdkWindowsDesktop)' != 'true'" 
-                   ResourceName="UseWpfOrUseWindowsFormsRequiresWindowsDesktopFramework" />
+  <Target Name="_CheckForTransitiveWindowsDesktopDependencies"
+          AfterTargets="ResolvePackageAssets"
+          Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0')) and '$(TargetPlatformIdentifier)' != 'Windows' and '@(TransitiveFrameworkReference)' != ''">
+    <NetSdkError Condition="@(TransitiveFrameworkReference->'%(Identity)'->Contains('Microsoft.WindowsDesktop.App'))"
+                 ResourceName="WindowsDesktopTargetPlatformMustBeWindows" />
+  </Target>
+
+  <Target Name="_CheckForInvalidWindowsDesktopTargetingConfiguration"
+        BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+        Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0')) and ('$(TargetPlatformIdentifier)' == 'Windows' or '$(UseWindowsForms)' == 'true' or '$(UseWPF)' == 'true')">
+    <NETSdkError Condition="('$(UseWindowsForms)' == 'true' or '$(UseWPF)' == 'true') and '$(TargetPlatformIdentifier)' != 'Windows'"
+                 ResourceName="WindowsDesktopTargetPlatformMustBeWindows" />
+  </Target>
+
+  <Target Name="_CheckForUnnecessaryWindowsDesktopSDK"
+        BeforeTargets="_CheckForInvalidConfigurationAndPlatform"
+        Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0')) and '$(MicrosoftWindowsDesktopSdkPath)' != ''">
```

Comment:
Looking at the WindowsDesktop Sdk.props file, it looks like the MicrosoftWindowsDesktopSdkPath property is a test hook.  It would probably be better to rely on something else.  To do so we might need to add a property such as MicrosoftNETSdkWindowsDesktopImported to the WindowsDesktop Sdk.props.

I'm fine with merging this as is for now though.

Reply:
Cool, I've updated to use the `_MicrosoftWindowsDesktopSdkImported` property they recently added.